src/Service/OpenamApiClient.php

```diff
@@ -150,6 +160,28 @@ public function cleanSubjectId($string) {
     return str_replace([' ', '"'], ['+', ''], $string);
   }
 
+  /**
+   * Encode a string for MIME header.
+   *
+   * This method is used for username and password headers.
+   * HTTP headers only support ISO-8859-1 characters. Therefore OpenAM allow
+   * username and password headers to be MIME encoded.
+   *
+   * Use this method instead of mb_encode_mimeheader() as the latter can not
+   * handle large strings, it also will not encode email addresses.
+   *
+   * https://bugster.forgerock.org/jira/browse/OPENAM-3750
+   *
+   * @param string $header_value
+   *   The string to be encoded.
+   *
+   * @return string
+   *   The encoded string.
+   */
+  public function base64EncodeHeader($header_value) {
+    return '=?UTF-8?B?' . base64_encode($header_value) . '?=';
```

Comment:
would we be able to use `Crypt::hashBase64`?

Reply:
I don't think we could as it would also hash the data. OpenAM wants it to be UTF8 MIME encoded, we only create our version of mb_encode_mimeheader as it doesn't handled email address correctly. However, all MIME encoding really is, is a base 64 coding something and adding a charset so we do this manually instead.

Full details of this are on Forgerock's Jira https://bugster.forgerock.org/jira/browse/OPENAM-3750?focusedCommentId=81877#comment-81877